### DISCUSSION TITLE
Cloudbet GraphQL Betting API

### DISCUSSION THREAD

**Discussion by kgravenreuth** [cloudbet/docs#29] [Show and tell]
Hey guys, we've launched a GraphQL API for Cloudbet which allows you to query our API feed, place bets and check account details from a single API endpoint. You get automatic documentation about all our API endpoints in one place too. Pretty cool!

Technical blog post with detailed examples about the [Cloudbet GraphQL Betting API](https://www.cloudbet.com/en/blog/posts/sports-betting-graphql-api-with-javascript-node-js?utm_source=github.com&utm_medium=social&utm_campaign=axis)

Both the GraphQL API documentation and GraphQL API endpoints are available at [https://sports-api-graphql.cloudbet.com/graphql](https://sports-api-graphql.cloudbet.com/graphql?utm_source=github.com&utm_medium=social&utm_campaign=axis)

It's an Alpha API at present, so there might be rough edges. Let us know what you think. 

We will also open source the GraphQL API implementation in the next few days. Stay tuned. 

Cheers,
Klaus Gravenreuth
Product & Engineering @ Cloudbet

**Comment by ghost** [comment]
Hi! @kgravenreuth Great news. I was wondering, if I place bets through here - will my bets appear on the website UI? If not - are there any plans in the future?

**Reply by kgravenreuth** [reply]
> Hi! @kgravenreuth Great news. I was wondering, if I place bets through here - will my bets appear on the website UI? If not - are there any plans in the future?

Hi @sonderangebot10 

Glad to hear that you're trying our GraphQL API!

For Public Trading API betting (both via REST and GraphQL), the bets are available only via the bet history endpoints listed at https://www.cloudbet.com/api/?urls.primaryName=Trading :

1. `/v3/bets/{referenceId}/status`
2. `/v4/bets/history`

These bets are not listed in the UI as the Bet history in the UI is meant for bets placed via the website. This is to avoid any confusion your part about whether the bets were placed via the UI or via the API.

Please let us know if you have any further questions.

Cheers,
Klaus

**Comment by Alicedale101** [comment]
Is there a endpoint to retrieve odds for all the sports, i want to say build a bot that follows my teams and if the odds say are 1.6 for my team then i would like the bot to make a automatically bet for me is that possible

**Reply by kgravenreuth** [reply]
Hi @Alicedale101 

Please refer to https://github.com/Cloudbet/docs/discussions/68 and we can continue the discussion there, cheers!

**Comment by kgravenreuth** [comment]
FYI, we’re no longer supporting the hosted GraphQL gateway.
We’ve open-sourced it here: https://github.com/cloudbet/graphql-gateway

If you still need it, clone the repo and host it yourself.
